File: LeetCode-Java/Excel Sheet Column Number.java
public class Solution {
    public int titleToNumber(String s) {
        int result = 0;
        int unit = 1;
        for(int i = s.length() -1 ;i >= 0;i --)
        {
            result += (s.charAt(i) - 'A' + 1) * unit;
            unit *= 26;
        }
        return result;
    }
}

File: Leetcode-cpp/excel_sheet_column_number.cpp
class Solution {
 public:
  int titleToNumber(string s) {
    int result_raw = 0;
    int result_offset = 0;
    int offset = 1;
    for (int i = 0; i < s.length(); ++i) {
      result_raw = 26 * result_raw + s[i] - 'A';
      result_offset += offset;
      offset *= 26;
    }
    return result_raw + result_offset;
  }
};